transport/tlsfrag/buffer.go

```diff
@@ -89,8 +85,12 @@ func (b *clientHelloBuffer) ReadFrom(r io.Reader) (n int64, err error) {
 		n += int64(m)
```

Comment:
We should comment somewhere that the Read above will never return if less than 5 bytes is sent (e.g. "PING"). That's something we need to be aware of.

Reply:
Sure. And I was thinking of this kind of hold-and-wait issue today, maybe we can have something like `WithWaitTimeout(400 * time.Milliseconds)`. In that case, we will hold the header until either it's fulfilled by a `Write` or `ReadFrom` within `0.4s`, or timeout; if timeout occurs, we won't do fragmentation and simply forwards the traffic to the `base` conn.

Reply:
We can have a default value (either `2*rtt`, or the default TLS handshake timeout value like `10s`), we can also have an optional `WithWaitTimeout` passed to `NewFixedLenStreamDialer(base, frag, WithWaitTimeout(func (rtt time.Duration) time.Duration { return rtt * 1.5 }))`.

If it's just for TLS client handshake, we don't need to reset on every `Read`, this is a one-time initial timeout (because if it's a valid TLS client handshake, it must send all headers within a very short amount of time). Once it timed-out, we can simply fallback to non-frag behavior.